rest/data/generate.go

```diff
@@ -25,28 +27,21 @@ func (gc *GenerateConnector) GenerateTasks(taskID string, jsonBytes []json.RawMe
 	}
 	g := model.MergeGeneratedProjects(projects)
 	g.TaskID = taskID
-	p, v, t, pm, err := g.NewVersion()
-	if err != nil {
-		return err
-	}
-	syntaxErrs, err := validator.CheckProjectSyntax(p)
-	if err != nil {
-		return err
-	}
-	if len(syntaxErrs) > 0 {
-		return gimlet.ErrorResponse{
-			StatusCode: http.StatusBadRequest,
-			Message:    fmt.Sprintf("project syntax is invalid: %s", validator.ValidationErrorsToString(syntaxErrs)),
+
+	for i := 0; i < retryLimit; i++ {
```

Comment:
This should probably retry with exponential backoff with jitter. See util/retry.go.

Reply:
Thinking about this more, I wonder if there server shouldn't have any retries at all, since the client will retry anyway if the server returns an error. The server can just error if it detects a race, and the client will retry.